tests/integration/frameworks/eleventy.test.mjs

```diff
@@ -0,0 +1,158 @@
+import { Buffer } from 'buffer'
+import path from 'path'
+import { fileURLToPath } from 'url'
+
+import fetch from 'node-fetch'
+import { afterAll, beforeAll, describe, test } from 'vitest'
+
+import { clientIP, originalIP } from '../../lib/local-ip.cjs'
+import { startDevServer } from '../utils/dev-server.cjs'
+
+// eslint-disable-next-line no-underscore-dangle
+const __dirname = path.dirname(fileURLToPath(import.meta.url))
+
+const context = {}
+
+beforeAll(async () => {
+  const server = await startDevServer({ cwd: path.join(__dirname, '__fixtures__/eleventy-site') })
+
+  context.server = server
+})
+
+afterAll(async () => {
+  const { server } = context
+  await server.close()
+})
+
+describe('eleventy', () => {
+  test('homepage', async (t) => {
+    const { url } = context.server
+    const response = await fetch(`${url}/`).then((res) => res.text())
+
+    t.expect(response.includes('Eleventy Site')).toBe(true)
+  })
+
+  test('redirect test', async (t) => {
+    const { url } = context.server
+    const response = await fetch(`${url}/something`, {
+      redirect: 'manual',
+    })
+    const { headers, status } = response
+
+    t.expect(status).toBe(301)
+    t.expect(headers.get('location').endsWith('/otherthing')).toBe(true)
+    t.expect(await response.text()).toEqual('Redirecting to /otherthing')
+  })
+
+  // TODO: un-skip this once https://github.com/netlify/cli/issues/1242 is fixed
+  test.skip('normal rewrite', async (t) => {
+    const { url } = context.server
+    const response = await fetch(`${url}/doesnt-exist`)
+    const { headers, status } = response
+    const body = await response.text()
+
+    t.expect(status).toBe(200)
+    t.expect(headers.get('content-type').startsWith('text/html')).toBe(true)
+    t.expect(body.includes('Eleventy Site')).toBe(true)
+  })
+
+  test('force rewrite', async (t) => {
+    const { url } = context.server
+    const response = await fetch(`${url}/force`)
+    const { headers, status } = response
+    const body = await response.text()
+
+    t.expect(status).toBe(200)
+    t.expect(headers.get('content-type').startsWith('text/html')).toBe(true)
+    t.expect(body.includes('<h1>Test content</h1>')).toBe(true)
+  })
+
+  test('functions rewrite echo without body', async (t) => {
+    const { host, port, url } = context.server
+    const response = await fetch(`${url}/api/echo?ding=dong`).then((res) => res.json())
+    const { 'x-nf-request-id': requestID, ...headers } = response.headers
+
+    t.expect(response.body).toBe(undefined)
+    t.expect(headers).toStrictEqual({
+      accept: '*/*',
```

Comment:
original test had `accept: 'application/json'` - what has changed here?

Reply:
node-fetch by default returns `accept: '*/*'`
I have set the related headers in order to match with the original tests headers, except for `user-agent`